### ISSUE TITLE
Failure to parse hostname in after-response script

### ISSUE THREAD

**Issue by devguydavid** [getinsomnia/insomnia#8049]
### Expected Behavior

Parsing of the response succeeds

### Actual Behavior

This error:

Unexpected Request Failure

The request failed due to an unhandled error:

Execute after-response script failed: Error: Error: new Url(): failed to parse hostname in url ${urlStr}

### Reproduction Steps

1. Add a GET request that returns the full URL in a json response body property. For example, a call that returns this:
```
{
  "url": "https://google.com"
}
```
2. Add a PUT request that uses the full URL by referencing the Response -> Body Attribute of "url" returned by the request in step 1 as its endpoint to call.
3. Call the two endpoints in sequential order.

### Is there an existing issue for this?

- [X] I have searched the [issue tracker](https://www.github.com/Kong/insomnia/issues) for this problem.

### Additional Information

Here is a collection that exhibits the error using mocky.io:

[Reproduce Hostname Parsing Error.json](https://github.com/user-attachments/files/17265390/Reproduce.Hostname.Parsing.Error.json)

### Insomnia Version

10.0.0

### What operating system are you using?

macOS

### Operating System Version

Mac OS Sequoia 15.0

### Installation method

dmg downloaded from the website

**Comment by devguydavid** [comment]
It looks like `parse` in `urls.ts` is being passed the raw value instead of the resolved URL:

```
{% response 'body', 'req_3a1b3b4368a14cd59b53656d37022415', 'b64::JC51cmw=::46b', 'never', 60 %}
```

**Comment by subnetmarco** [comment]
@ihexxa

**Comment by ihexxa** [comment]
Hello @devguydavid, you are seeing `{% ... %}` because template tags (such as your case or environment tags) are evaluated after the pre-request script step. Probably you could try `insomnia.environment.set('url', url)` and access it in the next request at the moment and we will take a look at it, as currently there seems no quick and elegant way to handle this.

**Comment by devguydavid** [comment]
Yes, thank you, that is a fine workaround. One of the things I love about Insomnia is not needing to use environment variables for passing data around (looking at you, Postman), but in a pinch it works!
